directus-solidstart/index.md

```diff
@@ -0,0 +1,314 @@
+
+## Before You Start
+
+You will need:
+
+- To install Node.js and a code editor on your computer.
+- A Directus project - follow our quickstart guide if you don't already have one.
+- SolidJS knowledge will definitely help, but will not prevent you from following this guide.
+
+## Initialize SolidStart Project
+
+Open your terminal and run the following commands to create a new Solid project and add the Directus JavaScript SDK:
+
+```sh
+npm create solid@latest solid-directus
+```
+
+A prompt will start, here are the sequence of answers provided for this exercise:
+
+```text
+✔ Which template do you want to use?
+  › bare
+
+✔ Server Side Rendering?
+  › yes
+
+✔ Use TypeScript?
+  › no
+```
+
+Now change directory to what you created and install all dependencies while adding the Directus SDK.
+
+```sh
+cd solid-directus && npm install @directus/sdk
+```
+
+With this, we are ready to start coding!
+
+```sh
+npm run dev --open
+```
+
+It will open a browser window or tab on [localhost:3000](https://localhost:3000) or the next available port.
+
+## Create a Directus Helper
+
+We now need to setup the Directus SDK. We will add our data fetchers in this same file so we can handle caching and requests all at the same place. Create a new file `directus.js` inside of the src/libs directory:
+
+```jsx
+import { createDirectus, readItem, readItems, rest } from "@directus/sdk";
+import { cache } from "@solidjs/router";
+
+export const directus = createDirectus("< your project URL >").with(rest());
+```
+
+We have some unused `import` statements at the moment since now we are only instantiating the Directus client.
+
+> Remember to replace `"< your project URL >"` with your actual URL, you can find it in your Directus dashboard.
+
+It may be useful to create an `.env` file and store the URL there in case you use different projects for production, development, etc. Otherwise it is not a private value, access is managed through the Directus dashboard.
+
+## Using Global Metadata and Settings
+
+In your Directus project, navigate to Settings -> Data Model and create a new collection called **global**. Under the Singleton option, select 'Treat as a single object', as this collection will have just a single entry containing global website metadata.
+
+![A form named "Global" has two inputs - a title and a description, each filled with some text.](solid-global-config.webp)
+
+By default, new collections are not accessible to the public. Navigate to Settings -> Roles & Permissions -> Public and give Read access to the Global collection.
+
+## Fetch Global Data
+
+Now that the table is created and we have data added, let's create our first fetcher. In `/lib/directus.js` we will add a new method:
+
+```js
+export const getGlobals = cache(async () => {
+  "use server";
+  try {
+    return await directus.request.readItems("global");
+  } catch (e) {
+    console.error(e);
+
+    return null;
+  }
+}, `globals`);
```

Comment:
```suggestion
}, "globals");
```

Wouldn't normally bother me but you wrap in double quotes below and immediately I couldn't match up what you wrote with the code because of the differing quote types.